[kevin-pek] iP

kevin-pek opened the pull request:
# Skibidi Chatbot

Skibidi chatbot is a **free** chatbot to help you remember the things you need to do in case your brain rots.

- text-based
- easy to learn
- ~fast~ 🔥🔥🔥 *BLAZINGLY FAST*

To try out this app, follow these steps:

1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
2. double-click it
3. add your tasks

>"deep sounding quote" ~ important person name, ([for more quotes](https://inspirobot.me/))

## Roadmap

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

## Code snippets

You might encounter a significant amount of frustration when using the app since it is currently 100% text based which could lead to suboptimal user experience. To worsen this experience, create and run a file names `Main.java` with the following code:

```java
import java.io.IOException;

public class Main {
	public static void main(String[] args) throws IOException {
		Runtime.getRuntime().exec(new String[]{"java", "-cp", System.getProperty("java.class.path"), "Main"});
	}
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kevin-pek committed:
Level-0: Rename, Greet, Exit

kevin-pek committed:
Level-1: Echo

kevin-pek committed:
Level-2: Add, List

kevin-pek committed:
Level-3: Mark as done

kevin-pek committed:
Level-4: ToDos, Events, Deadlines

kevin-pek committed:
A-TextUiTesting: Add package and maven configuration

kevin-pek committed:
Update .gitignore

kevin-pek committed:
Level-5: Handle Errors

kevin-pek committed:
Level-6: Delete

kevin-pek committed:
Modify exception type thrown in event command

kevin-pek committed:
Change command parsing to use enum, improve error handling

kevin-pek committed:
Fix input validation for mark, unmark, delete commands

kevin-pek committed:
Refactor Task class to AbstractTask

kevin-pek committed:
Refactor command error handling in Skibidi

kevin-pek committed:
Add data folder for tasks

kevin-pek committed:
Refactor classes into separate packages

kevin-pek committed:
Merge branch 'branch-Level-7'

kevin-pek committed:
Convert Deadline and Event fields to LocalDate

kevin-pek committed:
Add date parsing for deserialization function

kevin-pek committed:
Refactor Skibidi command parsing and execution block

kevin-pek committed:
Merge branch 'branch-Level-8'

kevin-pek committed:
Refactor code to meet style guide

kevin-pek committed:
Add find tasks command

kevin-pek committed:
Add input validation for find command

kevin-pek committed:
Merge branch 'branch-Level-9'

kevin-pek committed:
Add javafx and gui for application

kevin-pek committed:
Merge branch 'branch-Level-10'

kevin-pek committed:
Merge branch 'add-gradle-support'

kevin-pek committed:
Add checkstyle to build.gradle

kevin-pek committed:
Merge branch 'branch-A-CheckStyle'

kevin-pek committed:
Add javadoc comments for public functions

kevin-pek committed:
Merge branch 'branch-A-JavaDoc'

somethingfishyfishy commented on src/main/java/skibidi/Skibidi.java:
```diff
@@ -0,0 +1,71 @@
+import gui.MainGui;
+import javafx.application.Application;
+
+/**
```
(not a violation)
Just pointing out for single liners, you could write the javadoc comment as a single line.

Eg.
/** Number of connections to this database */
private int connectionCount;

kevin-pek replied:
Thanks for the input! This will help make the codebase more concise and readable.

somethingfishyfishy commented on src/main/java/skibidi/Storage.java:
```diff
@@ -0,0 +1,73 @@
+            } else {
+                System.out.println("Failed to save changes to file.");
+            }
+        } catch (TaskList.TaskNotFoundException err) {
```
can apply polymorphism.

Eg.
catch (Exception e) {System.out.println(e.getMessage())}

kevin-pek replied:
Good call, a generic Exception handler would suffice for cases where special handling is not needed.

LimKaiWei commented on src/main/java/skibidi/command/AbstractCommand.java:
interesting choice of class name, many just use `Command`

kevin-pek replied:
Right, I added this prefix for abstract classes so it would be clearer at a glance that this was an abstract class. Though I see why people would find this unnecessary too. Going to leave it as it is since it doesn't go against any of the coding standards (to my knowledge).

LimKaiWei commented on src/main/java/skibidi/Storage.java:
```diff
@@ -0,0 +1,73 @@
+    /**
+     * Save given task list to storage data path. Overrides any existing data on the file.
+     */
+    public void saveTasksToDisk(TaskList taskList) {
```
i like this descriptive method name, here and throughout your code

kevin-pek replied:
Thanks!

kevin-pek committed:
Update print statements for various commands

kevin-pek committed:
Merge branch 'branch-A-CodingStandard'

kevin-pek committed:
Fix merge conflict for git branch

kevin-pek committed:
Refactor code comments for better compactness

kevin-pek committed:
Refactor find command to use java streams API

kevin-pek committed:
Merge branch 'branch-A-Streams'

kevin-pek committed:
Refactor command execution method to return String

kevin-pek committed:
Fix coding standard issue

kevin-pek committed:
Merge branch 'branch-A-CodeQuality'

kevin-pek committed:
Add unit tests for Todo and AbstractTask deserialization

kevin-pek committed:
Merge branch 'branch-A-JUnit'

kevin-pek committed:
Add custom exception for Task creation validation logc

kevin-pek committed:
Add assert statement for resources and add task method

kevin-pek committed:
Merge pull request #1 from kevin-pek/branch-A-Assertions

Branch-A-Assertions

kevin-pek committed:
Merge branch 'master' into branch-A-CodeQuality

kevin-pek committed:
Add github CI workflow configuration

kevin-pek committed:
Add SetPriorityCommand

kevin-pek committed:
Add priority score to AbstractTask class definition

kevin-pek committed:
Add command parsing for priority command

kevin-pek committed:
Reverse priority ordering for TaskList

kevin-pek committed:
Add getter for priority score and reduce priority visibility

kevin-pek committed:
Add priority score to toString method of AbstractTask

kevin-pek committed:
Merge pull request #3 from kevin-pek/branch-A-CI

Add github CI workflow configuration

kevin-pek committed:
Fix command parsing bug for todo

kevin-pek committed:
Merge branch 'master' into C-Priority

kevin-pek committed:
Fix command parsing for priority command

kevin-pek committed:
Merge pull request #4 from kevin-pek/C-Priority

Add priority score for AbstractTask

kevin-pek committed:
Merge branch 'master' into branch-A-CodeQuality

kevin-pek committed:
Refactor ui class methods

kevin-pek committed:
Merge pull request #2 from kevin-pek/branch-A-CodeQuality

Merge branch branch-A-CodeQuality into 'master'

kevin-pek committed:
MainGUI.java: Add scroll to bottom when new message is printed

Previously, the UI does not follow the new messages entered by the user
and the response that is printed by the chatbot.

This commit adds a listener to the height of the dialogContainer, to
scroll to the bottom of the page whenever it changes.

This improves the user experience and makes the app more similar to a
typical messaging app that users would be familiar with.

kevin-pek committed:
Skibidi.java: Add logic to update file every command

The tasks.txt file is not updated when running the chatbot
with javafx.

Move the saveTasksToDisk method from within the start method of
Skibidi.java to the getResponse method, which executes everytime a
command is entered by the user.

This automatically keeps the tasks.txt file up to date with the latest
task list.

kevin-pek committed:
Add dynamic resizing for gui nodes

Resizing the chatbot window does not change the size of the text input
or DisplayBox ui components.

Add more anchors to make sure that the size of these components follow
the resized window.

kevin-pek committed:
Update README.md and added AI.md

Add documentation describing how AI tools were used in introducing
changes to this project.

kevin-pek committed:
Merge pull request #5 from kevin-pek/branch-A-BetterGui

A-BetterGui

kevin-pek committed:
Add acknowledgements for reused code

kevin-pek committed:
Add missing acknowledgements to DialogBox.java

kevin-pek committed:
Add user guide with screenshots of app

kevin-pek committed:
Update printing of find command results using streams

kevin-pek committed:
Merge branch 'branch-A-UserGuide'

kevin-pek committed:
Add dependencies for fat jar building

kevin-pek committed:
Improve error handling for command parsing

kevin-pek committed:
Update error message format for number arguments

kevin-pek committed:
Merge branch 'branch-A-MoreErrorHandling'
